Some new commands

mindfuckup opened the pull request:
Hi

Here are some new commands:
* column
* ex
* psql
* rev
* ss
* ssh-keygen
* tbl
* troff
* xmodmap

Some of them are Nagios plugins:
* check_by_ssh
* check_cups
* check_log
* check_memory
* check_raid
* check_ssl_cert
* check_statusfile

I'm not sure if this is done correctly. If no command execution is possible, should the e.g. file write or file read command still be added to the sudo secion? Maybe I did this not as you would expect:

Best,
Mänu

mindfuckup commented:
The CI checks failed somehow, I don't know why.

I also just noticed, that I put the wrong filename in `ssh-keygen.md` it should be `ssh-keyscan.md`.

cyrus-and commented:
Wow thanks! I'm going to review all those in the next days...

> should the e.g. file write or file read command still be added to the sudo secion?

Yeah we usually do that, we pick the *best* function (e.g., `shell` > `file-read`) and we add it as `sudo` and `suid` where appropriate.

> The CI checks failed somehow, I don't know why.

See the [details](https://travis-ci.org/github/GTFOBins/GTFOBins.github.io/builds/743176165):

```
_gtfobins/check_ssl_cert.md
  6:6       error    wrong indentation: expected 4 but found 5  (indentation)
  15:6      error    wrong indentation: expected 4 but found 5  (indentation)
  25:6      error    wrong indentation: expected 4 but found 5  (indentation)
```

> I also just noticed, that I put the wrong filename in ssh-keygen.md it should be ssh-keyscan.md.

Feel free to also commit the fixes, you can even rewrite the history if you want, since I'm not going to review this right away.

mindfuckup commented:
I did a cleanup.

Also added the following commands:
- comm
- csplit
- install
- join
- paste
- pr
- split

Added file read for:
- nroff

mindfuckup commented:
I hope I documented everything in the style you expect.

Maybe a comment when you review it: I was not always sure in the `suid` function when a shell is invoked. Especially in `ex` and `psql`. This might also be a `limited-suid`. Can you also have a detailed look into that?

THX & best regards,
Emanuel

cyrus-and commented:
Well... better late than never. :)

Thank you very much for this, you did a really good job in keeping up with our conventions!

I changed a few things, feel free to ask if it's not clear.

mindfuckup commented:
Nice, thank you for merging. Your changes make sense to me. I think I also got the differentiation between SUID and limited SUID.

cyrus-and commented:
> I think I also got the differentiation between SUID and limited SUID.

What I usually do to discern between SUID and limited SUID is to use `strace` with the command I want to check.

### Limited SUID

If run through a shell without the possibility of adding a `-p` (usually a `system()` invocation):

```
$ strace -qqqfe execve python -c 'import os; os.system("/usr/bin/id")'
execve("/usr/bin/python", ["python", "-c", "import os; os.system(\"/usr/bin/i"...], 0x7ffce120ac20 /* 14 vars */) = 0
[pid    76] execve("/bin/sh", ["sh", "-c", "/usr/bin/id"], 0x7ffc5ba7bce8 /* 14 vars */) = 0
[pid    77] execve("/usr/bin/id", ["/usr/bin/id"], 0x559f21a46b60 /* 14 vars */) = 0
uid=1000(user) gid=1000(user) groups=1000(user),27(sudo)
```

### SUID

If run directly:

```
$ strace -qqqfe execve env /usr/bin/id
execve("/usr/bin/env", ["env", "/usr/bin/id"], 0x7ffeda4c81f8 /* 14 vars */) = 0
execve("/usr/bin/id", ["/usr/bin/id"], 0x7ffdae8b24a0 /* 14 vars */) = 0
uid=1000(user) gid=1000(user) groups=1000(user),27(sudo)
```
